src/main/java/org/opensearch/neuralsearch/query/NeuralSparseQueryBuilder.java

```diff
@@ -161,16 +187,40 @@ public static NeuralSparseQueryBuilder fromXContent(XContentParser parser) throw
         }
 
         requireValue(sparseEncodingQueryBuilder.fieldName(), "Field name must be provided for " + NAME + " query");
-        requireValue(
-            sparseEncodingQueryBuilder.queryText(),
-            String.format(Locale.ROOT, "%s field must be provided for [%s] query", QUERY_TEXT_FIELD.getPreferredName(), NAME)
-        );
-        if (!isClusterOnOrAfterMinReqVersionForDefaultModelIdSupport()) {
+        if (Objects.isNull(sparseEncodingQueryBuilder.queryTokensSupplier)) {
             requireValue(
-                sparseEncodingQueryBuilder.modelId(),
-                String.format(Locale.ROOT, "%s field must be provided for [%s] query", MODEL_ID_FIELD.getPreferredName(), NAME)
+                sparseEncodingQueryBuilder.queryText(),
+                String.format(
+                    Locale.ROOT,
+                    "either %s field or %s field must be provided for [%s] query",
+                    QUERY_TEXT_FIELD.getPreferredName(),
+                    QUERY_TOKENS_FIELD.getPreferredName(),
+                    NAME
+                )
+            );
+            if (!isClusterOnOrAfterMinReqVersionForDefaultModelIdSupport()) {
+                requireValue(
+                    sparseEncodingQueryBuilder.modelId(),
+                    String.format(
+                        Locale.ROOT,
+                        "using %s, %s field must be provided for [%s] query",
+                        QUERY_TEXT_FIELD.getPreferredName(),
+                        MODEL_ID_FIELD.getPreferredName(),
+                        NAME
+                    )
+                );
+            }
+        }
+
+        if (Objects.nonNull(sparseEncodingQueryBuilder.queryText) && StringUtils.EMPTY.equals(sparseEncodingQueryBuilder.queryText)) {
```

Comment:
1. nit: use getter instead of reference to member variable.
2. If we only want to check sparseEncodingQueryBuilder.queryText() is empty string, `StringUtils.EMPTY.equals` is enough, if we want to check null or "", then we should use `||` instead

Reply:
ack